### PR TITLE
Correct documentation

### LINKED ISSUE #1611: Mistakes in documentation
Just filing these so that I don't forget:
- [Header guidelines](https://microsoft.github.io/reverse-proxy/articles/header-guidelines.html#traceparent-request-id-tracestate-baggage-correlation-context) docs mention all telemetry headers being removed. This is no longer the case with 1.1 so we need to rephrase that.
- Our configs samples use `"Version" : "2"` for the request version. We should make this explicitly `2.0` since changing it to just `1` will kill your perf.

### PR DESCRIPTION
**Pull request by greenEkatherine**
Fixes #1611

### CODE REVIEW

File: src/ReverseProxy/Configuration/ConfigValidator.cs [hunk 1]
```diff
@@ -510,4 +520,17 @@ private void ValidatePassiveHealthCheck(IList<Exception> errors, ClusterConfig c
...
     }
+
+    private static class Log
+    {
+        private static readonly Action<ILogger, Exception?> _requestVersionDetected = LoggerMessage.Define(
+            LogLevel.Warning,
+            EventIds.LoadData,
+            "HttpRequest version is set to 1.0, for better performance please set up the higher version.");
+
+        public static void ResponseReceived(ILogger logger)
```
Comment (Tratcher) [inline src/ReverseProxy/Configuration/ConfigValidator.cs:531]:
```suggestion
        public static void Http10Version(ILogger logger)
```

File: src/ReverseProxy/Configuration/ConfigValidator.cs [hunk 2]
```diff
@@ -510,4 +520,17 @@ private void ValidatePassiveHealthCheck(IList<Exception> errors, ClusterConfig c
             errors.Add(new ArgumentException($"Unhealthy destination reactivation period set on the cluster '{cluster.ClusterId}' must be positive."));
         }
     }
+
+    private static class Log
+    {
+        private static readonly Action<ILogger, Exception?> _requestVersionDetected = LoggerMessage.Define(
+            LogLevel.Warning,
+            EventIds.LoadData,
+            "HttpRequest version is set to 1.0, for better performance please set up the higher version.");
```
Comment (Tratcher) [inline src/ReverseProxy/Configuration/ConfigValidator.cs:529]:
```suggestion
            "The HttpRequest version is set to 1.0 which can result in poor performance and port exhaustion. Use 1.1, 2, or 3 instead.");
```

**Review by Tratcher — APPROVED** [review]

File: src/ReverseProxy/Configuration/ConfigValidator.cs [hunk 3]
```diff
@@ -434,6 +438,12 @@ private static void ValidateProxyHttpRequest(IList<Exception> errors, ClusterCon
         {
             errors.Add(new ArgumentException($"Outgoing request version '{cluster.HttpRequest.Version}' is not any of supported HTTP versions (1.0, 1.1, 2 and 3)."));
         }
+
+        if (cluster.HttpRequest.Version is not null &&
+            cluster.HttpRequest.Version == HttpVersion.Version10)
```
Comment (MihaZupan) [inline src/ReverseProxy/Configuration/ConfigValidator.cs:443]:
```suggestion
        if (cluster.HttpRequest.Version == HttpVersion.Version10)
```

**Review by MihaZupan — APPROVED** [review]

File: src/ReverseProxy/Configuration/ConfigValidator.cs [hunk 4]
```diff
@@ -510,4 +519,17 @@ private void ValidatePassiveHealthCheck(IList<Exception> errors, ClusterConfig c
             errors.Add(new ArgumentException($"Unhealthy destination reactivation period set on the cluster '{cluster.ClusterId}' must be positive."));
         }
     }
+
+    private static class Log
+    {
+        private static readonly Action<ILogger, Exception?> _requestVersionDetected = LoggerMessage.Define(
+            LogLevel.Warning,
+            EventIds.LoadData,
```
[the remark this answers is not in the document]
Comment (greenEkatherine) [reply src/ReverseProxy/Configuration/ConfigValidator.cs:527]: Ok, will add

**Pull request merged**